website/docs/table-design/table-types/pk-table/merge-engines/aggregation.md

```diff
@@ -0,0 +1,534 @@
+- **Null Handling**: Null values are ignored
+
+**Example:**
+```java
+Schema schema = Schema.newBuilder()
+    .column("id", DataTypes.BIGINT())
+    .column("has_all_permissions", DataTypes.BOOLEAN(), AggFunction.BOOL_AND)
+    .primaryKey("id")
+    .build();
+
+TableDescriptor.builder()
+    .schema(schema)
+    .property("table.merge-engine", "aggregation")
+    .build();
+
+// Input: (1, true), (1, true), (1, false)
+// Result: (1, false) -- Not all values are true
+```
+
+### bool_or
+
+Checks if at least one boolean value in a set is true (logical OR).
+
+- **Supported Data Types**: BOOLEAN
+- **Behavior**: Returns true if any value is true
+- **Null Handling**: Null values are ignored
+
+**Example:**
+```java
+Schema schema = Schema.newBuilder()
+    .column("id", DataTypes.BIGINT())
+    .column("has_any_alert", DataTypes.BOOLEAN(), AggFunction.BOOL_OR)
+    .primaryKey("id")
+    .build();
+
+TableDescriptor.builder()
+    .schema(schema)
+    .property("table.merge-engine", "aggregation")
+    .build();
+
+// Input: (1, false), (1, false), (1, true)
+// Result: (1, true) -- At least one value is true
+```
+
+## Advanced Configuration
+
+### Default Aggregate Function
+
+You can set a default aggregate function for all non-primary key fields that don't have an explicitly specified function:
+
+```java
+Schema schema = Schema.newBuilder()
+    .column("id", DataTypes.BIGINT())
+    .column("col1", DataTypes.STRING())  // Defaults to LAST_VALUE_IGNORE_NULLS
+    .column("col2", DataTypes.BIGINT(), AggFunction.SUM)  // Explicitly set to SUM
+    .primaryKey("id")
+    .build();
+
+TableDescriptor.builder()
+    .schema(schema)
+    .property("table.merge-engine", "aggregation")
+    .build();
+```
+
+In this example:
+- `col2` uses `sum` aggregation (explicitly specified)
+- `col1` uses `last_value_ignore_nulls` as the default
+
+### Partial Update with Aggregation
+
+The aggregation merge engine supports partial updates through the UpsertWriter API. When performing a partial update:
+
+- **Target columns**: These columns will be aggregated according to their configured aggregate functions
+- **Non-target columns**: These columns will retain their existing values from the old row
+
+**Example:**
+
+```java
+Schema schema = Schema.newBuilder()
+    .column("id", DataTypes.BIGINT())
+    .column("count1", DataTypes.BIGINT(), AggFunction.SUM)
+    .column("count2", DataTypes.BIGINT(), AggFunction.SUM)
+    .column("sum1", DataTypes.DOUBLE(), AggFunction.SUM)
+    .column("sum2", DataTypes.DOUBLE(), AggFunction.SUM)
+    .primaryKey("id")
+    .build();
+
+TableDescriptor tableDescriptor = TableDescriptor.builder()
+    .schema(schema)
+    .property("table.merge-engine", "aggregation")
+    .build();
+
+// Create partial update writer targeting only id, count1, and sum1
+int[] targetColumns = new int[]{0, 1, 3}; // id, count1, sum1
+UpsertWriter partialWriter = table.newUpsert()
+    .withPartialUpdate(targetColumns)
```

Comment:
Use name-based `.partialUpdate("id", "count1", "sum1")` which is more straightforward.